datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java

```diff
@@ -83,15 +83,13 @@ public CustomOidcAuthenticator(final OidcConfiguration configuration, final Oidc
     }
 
     final ClientID _clientID = new ClientID(configuration.getClientId());
-    if (ClientAuthenticationMethod.NONE.equals(chosenMethod)) {
-      clientAuthentication = new EmptyAuthentication(_clientID);
-    } else if (ClientAuthenticationMethod.CLIENT_SECRET_POST.equals(chosenMethod)) {
+    if (ClientAuthenticationMethod.CLIENT_SECRET_POST.equals(chosenMethod)) {
       final Secret _secret = new Secret(configuration.getSecret());
       clientAuthentication = new ClientSecretPost(_clientID, _secret);
     } else if (ClientAuthenticationMethod.CLIENT_SECRET_BASIC.equals(chosenMethod)) {
       final Secret _secret = new Secret(configuration.getSecret());
       clientAuthentication = new ClientSecretBasic(_clientID, _secret);
-    } else {
+    } else if (!ClientAuthenticationMethod.NONE.equals(chosenMethod)) {
```

Comment:
can we explicitly handle none to make reading a bit easier? 

for example, 

else if (ClientAuthenticationMethod.NONE.equals(chosenMethod)) {
   clientAuthentication = null; // No client authentication in none mode. 
}